application/controllers/IcingadblistController.php

```diff
@@ -0,0 +1,169 @@
+                $this->addControl($searchBar);
+                $this->sendMultipartUpdate();
+                return;
+            }
+        } else {
+            $filter = $searchBar->getFilter();
+        }
+
+        $this->filter($hosts, $filter);
+
+        $this->addControl($paginationControl);
+        $this->addControl($sortControl);
+        $this->addControl($limitControl);
+        $this->addControl($searchBar);
+        $this->handleTimeRangePickerRequest();
+        $this->addControl(HtmlString::create($this->renderTimeRangePicker($this->view)));
+
+        $this->addContent((new IcingadbGraph($hosts)));
+
+        $this->setAutorefreshInterval(10);
+    }
+
+    public function servicesAction()
+    {
+        $this->setTitle(t('Services'));
+
+        $db = $this->getDb();
+
+        $services = Service::on($db)->with(['state']);
+        //$services->setResultSetClass(VolatileStateResults::class);
+        $services->getSelectBase()
+            ->where(['performance_data != ?' => '']);
+
+        $this->applyRestrictions($services);
+        $this->handleSearchRequest($services);
+
+        TimeRangePickerTools::copyAllRangeParameters(
+            Url::fromPath('icingadb/host')->getParams(),
+            $this->getRequest()->getUrl()->getParams()
+        );
+
+        TimeRangePickerTools::copyAllRangeParameters(
+            Url::fromPath('icingadb/service')->getParams(),
+            $this->getRequest()->getUrl()->getParams()
+        );
+
+        $limitControl = $this->createLimitControl();
+        $paginationControl = $this->createPaginationControl($services);
+        $sortControl = $this->createSortControl($services, [
+            'service.display_name' => t('Servicename'),
+            'host.display_name' => t('Hostname')
+        ]);
+
+        $searchBar = $this->createSearchBar($services, [
+            $limitControl->getLimitParam(),
+            $sortControl->getSortParam()
+        ]);
+
+        if ($searchBar->hasBeenSent() && ! $searchBar->isValid()) {
+            if ($searchBar->hasBeenSubmitted()) {
+                $filter = $this->getFilter();
+            } else {
+                $this->addControl($searchBar);
+                $this->sendMultipartUpdate();
+                return;
+            }
+        } else {
+            $filter = $searchBar->getFilter();
+        }
+
+        $this->filter($services, $filter);
+
+        $this->addControl($paginationControl);
+        $this->addControl($sortControl);
+        $this->addControl($limitControl);
+        $this->addControl($searchBar);
+        $this->handleTimeRangePickerRequest();
+        $this->addControl(HtmlString::create($this->renderTimeRangePicker($this->view)));
+
+        $this->addContent((new IcingadbGraph($services)));
+
+        $this->setAutorefreshInterval(10);
+    }
+
+    public function completeAction()
+    {
+        $suggestions = new ObjectSuggestions();
+        $suggestions->setModel(Service::class);
```

Comment:
This way it's only for services. For hosts, invalid suggestions may be the result.

Reply:
It may work, but it's still wrong.

>  Should I separate host and service actions? (Create a new class for each with a complete action for each).

Sounds like a good idea! You could also make these the default routes without the need to blacklist anything. The menu entries can be dynamically provided after all.